### PR TITLE
Fetch image data over the web

### PR DESCRIPTION
**Pull request by rprimet**
A few comments:

- not sure about the subfolder when DICODILE_DATA_HOME is set (e.g. `/home/user/dicodile/dicodile`) but it seems to match libsvmdata's behavior
- I did not update the hubble images for now, as it fails with what seems to be a certificate verification error (using `wget` or the `requests` library to download the images manually fails with an SSL check error on two distinct machines)
- Not sure about the mandril unit test, as fetching an image in a unit test does not seem very valuable (I mostly added it to make codecov happy)

### CODE REVIEW

File: dicodile/data/images.py [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
-
-DATA_DIR = os.environ.get("DATA_DIR", "../../data")
-
+from .home import DATA_HOME
 
 def get_mandril():
 
-    mandril = os.path.join(DATA_DIR,
-                           "images/standard_images/mandril_color.tif")
+    mandril_dir = pathlib.Path(DATA_HOME) / "images" / "standard_images"
```
Comment (tomMoral) [inline dicodile/data/images.py:11]: It is already a `Path` so I would say it is safe to do:
```suggestion
    mandril_dir = DATA_HOME / "images" / "standard_images"
```

File: dicodile/data/images.py [hunk 2]
```diff
@@ -1,15 +1,17 @@
...
+    mandril_dir.mkdir(parents=True, exist_ok=True)
+    mandril = download("http://sipi.usc.edu/database/download.php?vol=misc&img=4.2.03", mandril_dir / "mandril_color.tif")
```
Comment (tomMoral) [inline dicodile/data/images.py:13]: pep8:
```suggestion
    mandril = download(
        "http://sipi.usc.edu/database/download.php?vol=misc&img=4.2.03",
        mandril_dir / "mandril_color.tif"
    )
```

File: dicodile/data/tests/test_home.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        "XDG_DATA_HOME": None
+    })  
+    assert(get_data_home() == Path.home() / "data" / "dicodile")
+
+def _set_env(monkeypatch, d):
+    for k,v in d.items():
+        if v is not None:
+            monkeypatch.setenv(k,v)
+        else:
+            monkeypatch.delenv(k, raising=False)
```
Comment (tomMoral) [inline dicodile/data/tests/test_home.py:37]: This is very nice! Thanks for the `monkeypatch` trick, I will reuse it elsewhere!

**Review by tomMoral — APPROVED** [review]
A few nitpicks but overall this is very neat! Thx @rprimet  :)

File: dicodile/data/images.py (hunk 1 above)
Comment (rprimet) [reply dicodile/data/images.py]: good catch, thanks!

File: dicodile/data/images.py [hunk 4]
```diff
@@ -18,7 +23,7 @@ def get_hubble(size="Medium"):
 
     image_path = f"images/hubble/STScI-H-2016-39-a-{size}.jpg"
 
-    image_path = os.path.join(DATA_DIR, image_path)
+    image_path = os.path.join(DATA_HOME, image_path)
```
Comment (tomMoral) [inline dicodile/data/images.py:26]:
```suggestion
    image_path = DATA_HOME / image_path
```

File: dicodile/data/images.py [hunk 5]
```diff
@@ -1,15 +1,20 @@
 import os
```
Comment (tomMoral) [inline dicodile/data/images.py:1]: You can probably remove it.
```suggestion
```

**Review by tomMoral — APPROVED** [review]
Just 2 leftover nitpicks.

File: dicodile/data/images.py (hunk 4 above)
Comment (rprimet) [reply dicodile/data/images.py:26]: good point, thanks!

**Pull request merged**